### PR TITLE
[fix] Device and format and implementation optimization

### PR DESCRIPTION
**Pull request by KexinFeng**
1. Add the space char to the output
2. Add the test script of "huggyllama/llama-7b" for rolling_batch scheduler
3. Device setting is fixed. The following way will involve model copying, and cause OOM on huggyllama/llama-7b on 5.12xlarge.
```python
        if self.device:
            self.model.to(self.device)
```
4. Efficiency optimization by avoiding `list.pop(idx)`

### CODE REVIEW

File: engines/python/setup/djl_python/rolling_batch/scheduler_rolling_batch.py [hunk 1]
```diff
@@ -155,7 +160,7 @@ def _prefill_and_decode(self, new_requests):
         for request_id, generated_token, request in zip(
                 request_ids, generated_tokens, self.pending_requests):
             is_last_token = (request_id in exit_req_ids)
-            request.set_next_token(generated_token,
+            request.set_next_token(" " + generated_token,
```
Comment (frankfliu) [inline engines/python/setup/djl_python/rolling_batch/scheduler_rolling_batch.py:163]:
```suggestion
            request.set_next_token(f" {generated_token}",
```

File: .gitignore [hunk 2]
```diff
@@ -22,9 +22,13 @@ libs
 .classpath
 bin/
 
-#node
+# node
```
Comment (lanking520) [inline .gitignore:25]: please fix this

File: .gitignore [hunk 3]
```diff
@@ -22,9 +22,13 @@ libs
...
 node_modules/
 
-#vscode
+# vscode
```
Comment (lanking520) [inline .gitignore:28]: fix this

File: .gitignore [hunk 4]
```diff
@@ -22,9 +22,13 @@ libs
...
 .vscode
+
+# dir
 tests/integration/models/
+engines/python/setup/djl_python/tests/kv_cache_demo
```
Comment (lanking520) [inline .gitignore:33]: are those generated now?

File: engines/python/setup/djl_python/rolling_batch/rolling_batch.py [hunk 5]
```diff
@@ -193,9 +193,9 @@ def postprocess_results(self, batch_size):
             res = {"data": req.get_next_token(), "last": req.is_last_token()}
             results.append(res)
 
-        for i in range(1, batch_size + 1):
-            if self.pending_requests[batch_size - i].is_last_token():
-                self.pending_requests.pop(batch_size - i)
+        self.pending_requests = [
+            req for req in self.pending_requests if not req.is_last_token()
```
Comment (lanking520) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch.py:197]: can you check if list order is maintained through this operations?

File: .gitignore (hunk 2 above)
Comment (lanking520) [reply .gitignore:25]: Not fixed?

**Review by lanking520 — APPROVED** [review]

**Pull request merged**